pennylane/transforms/intermediate_reps/parity_matrix.py

```diff
@@ -63,7 +73,25 @@ def parity_matrix(circ: qml.tape.QuantumScript, wire_order: Sequence = None):
         x_2: ─╭X─╰X─╰●─╭X─│──╰X─┤  x_2 ⊕ x_3
         x_3: ─╰●───────╰●─╰●────┤  x_3
 
+    For more details, see the `compilation page <https://pennylane.ai/compilation/parity-matrix-intermediate-representation>`__ on the parity matrix intermediate representation.
+
     """
+    if isinstance(circ, qml.workflow.QNode):
+
+        def wrapped(*args, **kwargs):
+            circ1 = qml.workflow.construct_tape(circ)(*args, **kwargs)
+            return parity_matrix(circ1, wire_order)
+
+        return wrapped
+
+    if callable(circ):
+
+        def wrapped2(*args, **kwargs):
+            circ1 = qml.tape.make_qscript(circ)(*args, **kwargs)
+            return parity_matrix(circ1, wire_order)
+
+        return wrapped2
```

Comment:
Transforms like this (see `adjoint_metric_tensor` as an example) are typically implemented as a tape transform, with a `@partial(transform, is_informative=True)` decorator. The `transform` API automatically handles the behaviour of the transform when it is called on tapes vs. qnodes.

Reply:
Note that this is not a `transform` that turns circuits to circuits, but rather circuits to parity matrices (/phase polynomial rep). My understanding was that in this case the `transform` decorator is not appropriate (and doesnt work)